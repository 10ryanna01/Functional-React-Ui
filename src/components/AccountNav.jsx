import { useState, useEffect, useRef } from "react";
import {
  IconAddUser,
  IconLogin,
  IconLogOut,
  IconSettings,
} from "../assets/Index";
import AccountSignUpStepper from "./AccountSignUpStepper";

export default function AccountNav({mobileNavMenu, setMobileNavMenu}) {
  const logUserInToggled = localStorage.getItem("user-logged-in") === "true";

  const [logUserIn, setLoguserIn] = useState(logUserInToggled);
  const [userSignUp, setUserSignUp] = useState(false);
  const [userAccontSettings, setAccontSettings] = useState(false);
  const [applyOverlay, setApplyOverlay] = useState(false);

  let userSignUpRef = useRef();

  useEffect(() => {
    localStorage.setItem("user-logged-in", logUserIn);
  }, [logUserIn]);

  useEffect(() => {
    let handleExitUserSignUp = (e) => {
      if (!userSignUpRef.current.contains(e.target)) {
        setUserSignUp(false); 
        setApplyOverlay(false);
      }
    };
    document.addEventListener("mousedown", handleExitUserSignUp);
    return () => {
      document.removeEventListener("mousedown", handleExitUserSignUp);
    };
  }, []);

  const handleLogIn = () => {
    setLoguserIn(true);
    setUserSignUp(false); 
  };
  const handleSignUpStepper = () => {
    setUserSignUp(true);
    
    setApplyOverlay(true);
  };
  const handleLogOut = () => {
    setLoguserIn(false);
  };

  return (
    <>
      <div className="UI__header__acc-nav">
        {!logUserIn ? (
          <div className="UI__header__acc-nav__list">
            <button
              className=" UI__utility__button-reset UI__header__acc-nav__list__item"
              onClick={handleSignUpStepper}
            >
              
              <IconAddUser className="UI__header__acc-nav__list__item__icon" />
              <h3 className="UI__header__acc-nav__list__item__copy" data-testid="AccNavtestC">sign up</h3>
            </button>

            <button className="UI__utility__button-reset" onClick={handleLogIn}>
              <IconLogin className="UI__header__acc-nav__list__item__icon" />
              <h3 className="UI__header__acc-nav__list__item__copy" data-testid="AccNavtestB">sign in</h3>
            </button>
          </div>
        ) : (
          <>
            <button className="UI__utility__button-reset">
              <IconSettings className="UI__header__acc-nav__list__item__icon" />
              <h3 className="UI__header__acc-nav__list__item__copy">
                {" "}
                settings{" "}
              </h3>
            </button>
            <button
              className="UI__utility__button-reset"
              onClick={handleLogOut}
            >
              <IconLogOut className="UI__header__acc-nav__list__item__icon" />
              <h3 className="UI__header__acc-nav__list__item__copy">
                {" "}
                log out{" "}
              </h3>
            </button>
          </>
        )}
      </div>
      <div data-testid="AccNavtestA" ref={userSignUpRef}>
        {userSignUp ? (
          <>
            <div className="UI__signup-form">
              <AccountSignUpStepper  setApplyOverlay={setApplyOverlay} setUserSignUp={setUserSignUp} mobileNavMenu={mobileNavMenu} setMobileNavMenu={setMobileNavMenu} />
            </div>
          </>
        ) : null}
      </div>

      {applyOverlay ? <div className="overlay" data-testid=""testD></div> : null}
    </>
  );
}
