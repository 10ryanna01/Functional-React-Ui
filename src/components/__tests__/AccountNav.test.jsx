import { render, screen, cleanup} from "@testing-library/react";
import renderer from  "react-test-renderer";
import '@testing-library/jest-dom/extend-expect';
import AccountNav from "../AccountNav";

afterEach(() =>{
    cleanup();
});
 

    test("displays signup nav items", () => {
        
        render(<AccountNav  />);
        const accNavElement = screen.queryByTestId  ("AccNavtestA");
        const accNavElementB = screen.queryByTestId  ("AccNavtestB");
        const accNavElementC = screen.queryByTestId  ("AccNavtestC");
        expect(accNavElement).toBeInTheDocument();
        expect(accNavElementB).toHaveTextContent("sign in");
        expect(accNavElementC).toHaveTextContent("sign up");
        expect(accNavElementC).not.toContainHTML("<h1>");
    }) ;

    test("matches snapshot", () => {
    const snapShotTestAccountNav = renderer.create(<AccountNav />).toJSON();
    console.log(snapShotTestAccountNav);
    expect(snapShotTestAccountNav).toMatchSnapshot();
    }) ;