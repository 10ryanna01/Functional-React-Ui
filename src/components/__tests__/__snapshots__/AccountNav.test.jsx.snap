// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="UI__header__acc-nav"
  >
    <div
      className="UI__header__acc-nav__list"
    >
      <button
        aria-label="sign up button"
        className=" UI__utility__button-reset UI__header__acc-nav__list__item"
        onClick={[Function]}
      >
        <svg
          className="UI__header__acc-nav__list__item__icon"
        >
          Icon-Add-User.svg
        </svg>
        <h3
          className="UI__header__acc-nav__list__item__copy"
          data-testid="AccNavtestC"
        >
          sign up
        </h3>
      </button>
      <button
        aria-label="sign in button"
        className="UI__utility__button-reset"
        onClick={[Function]}
      >
        <svg
          className="UI__header__acc-nav__list__item__icon"
        >
          Icon-Login.svg
        </svg>
        <h3
          className="UI__header__acc-nav__list__item__copy"
          data-testid="AccNavtestB"
        >
          sign in
        </h3>
      </button>
    </div>
  </div>,
  <div
    data-testid="AccNavtestA"
  />,
]
`;
